front-end/src/components/Post/VoteRefrendum.tsx

```diff
@@ -0,0 +1,178 @@
+
+import React, { useContext, useState } from 'react';
+import styled from '@xstyled/styled-components';
+import { /* Divider, */ Icon } from 'semantic-ui-react';
+import { ApiPromise } from '@polkadot/api';
+import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
+
+import { Form } from '../../ui-components/Form';
+import Button from '../../ui-components/Button';
+import { NotificationContext } from '../../context/NotificationContext';
+import { NotificationStatus } from '../../types';
+
+interface Props {
+	className?: string
+	referendumId?: number | null | undefined
+	api?: ApiPromise,
+	apiReady?: boolean,
+	getLinkedAccount: () => Promise<InjectedAccountWithMeta | undefined>
+}
+
+const VoteRefrendum = ({ className, api, apiReady, getLinkedAccount, referendumId }: Props) => {
+	const { queueNotification } = useContext(NotificationContext);
+
+	type ConvictionType = 'Locked1x' | 'Locked2x' | 'Locked3x' | 'Locked4x' | 'Locked5x' | 'Locked6x';
+	const [conviction, setConviction] = useState<ConvictionType>('Locked1x');
+
+	const onConvictionChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+		const convictionValue = event.currentTarget.value as ConvictionType;
+		setConviction(convictionValue);
+	};
+
+	const voteRefrendum = async (aye: boolean) => {
+		if (!api) {
+			return;
+		}
+
+		const linkedAccount = await getLinkedAccount();
+
+		if (!linkedAccount) {
+			return;
+		}
+
+		const vote = api.tx.democracy.vote(referendumId || 0, { aye, conviction });
```

Comment:
idem here, bailout if there's not referendumId. Voting on 0 won't work in 99.9% of the time :)

Reply:
bailing